my_plugin.py

```diff
@@ -5,6 +5,7 @@
 
 MENU_NAME = "ToolsMenu"  # no spaces in names, use CamelCase
 MENU_LABEL = "Tools"  # spaces are fine in labels
+MENU_ENTRY_NAME = "" # Store generated menu item, used when unregister
```

Comment:
looks good, only thing left is to change `MENU_ENTRY_NAME` to `__menu_entry_name` since it's a private variable, not a constant